include/revng/ADT/GenericGraph.h

```diff
@@ -1562,6 +1697,54 @@ struct GraphTraits<llvm::Inverse<T *>> {
   static T *getEntryNode(llvm::Inverse<T *> N) { return N.Graph; };
 };
 
+// TODO: We should move the `Undirected` definition to a separate header, and
+//       offer a default implementation for `GraphTraits<Undirected<T>>` for
+//       any `T` for which you have a definition of both `GraphTraits<T>` and
+//       `GraphTraits<Inverse<T>>`
+/// This class is used as a marker class to tell the graph iterator to treat the
+/// underlying graph as an undirected one
+template<class GraphType>
+struct Undirected {
+  const GraphType &Graph;
+
+  inline Undirected(const GraphType &G) : Graph(G) {}
+};
```

Comment:
I don't want to raise the ambitions of this PR too much, it's fine not to do this tied to `GenericGraph`, but could we at least move `llvm::Undirected` and `MutableEdgeNode::UndirectedChildIterator` to `revng/Support/Undirected.h`.

`MutableEdgeNode` is already a quite big `class` and `UndirectedChildIterator` doesn't seem to be tied to it in any way.

Reply:
As you suggested, I extracted the `llvm::Undirected` and `MutableEdgeNode::UndirectedChildIterator` in the new `Undirected.h` header. The only downside of this is the fact that the `GraphTraits` specialization for `Undirected` is still in `GenericGraph`, so things are bit more scattered in different points, but whatever